src/classes/ADDR_Seasonal_TEST.cls

```diff
@@ -0,0 +1,923 @@
+        
+        // Hack!  by setting our trigger handler as run,
+        // we can insert the new seasonal addresses but not have them processed.
+        // this way we can test the scheduled job!
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = true;
+        insert allNewAddrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+
+        //Schedule the job just for test coverage purposes. But we still have to manually run the
+        //batch, as batch jobs run from a scheduled job don't actually run after stopTest, as described
+        //at https://salesforce.stackexchange.com/questions/36806/does-test-stoptest-ensure-a-system-schedule-database-batchable-completes-in-a
+        STG_InstallScript.scheduleJobIfNotScheduled('Seasonal Addresses Update', '0 15 0 * * ?', 'ADDR_Seasonal_SCHED');
+        // Get the information from the CronTrigger API object
+        CronTrigger ct = [SELECT Id, CronExpression, TimesTriggered, NextFireTime FROM CronTrigger WHERE CronJobDetail.Name = 'Seasonal Addresses Update'];
+        // Verify the expressions are the same
+        System.assertEquals('0 15 0 * * ?', ct.CronExpression);
+        // Verify the job has not run
+        System.assertEquals(0, ct.TimesTriggered);
+        // Verify the next time the job will run
+        System.assert(String.valueOf(ct.NextFireTime).contains('00:15:00'));
+        
+        // run the scheduled batch directly
+        ADDR_Seasonal_BATCH batch = new ADDR_Seasonal_BATCH();
+        Test.startTest(); 
+        Database.executeBatch(batch);
+        Test.stopTest();
+        
+        // verify Account address is new!
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c 
+        from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assert(acc.BillingStreet.contains('New Seasonal Street'));
+            system.assert(acc.BillingCity.contains('New Seasonal City'));
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+        
+        // verify the previous addresses still are Default
+        accAddrs = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c, Latest_Start_Date__c, Latest_End_Date__c 
+                    from Address__c where ID in :accAddrs];
+        system.assertEquals(2, accAddrs.size());
+        for (Address__c addr : accAddrs) {
+            boolean fSeasonal = (addr.MailingStreet__c.contains('New Seasonal Street'));
+            system.assertEquals(!fSeasonal, addr.Default_Address__c);
+        }
+        
+        // verify Contact address is new!
+        map<Id, Contact> mapContactIdContact = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.